internals/templates/routes.js

```diff
@@ -4,22 +4,39 @@
 // They are all wrapped in the App component, which should contain the navbar etc
 // See http://blog.mxstbr.com/2016/01/react-apps-with-pages for more information
 // about the require.ensure code splitting business
+import { injectAsyncReducer } from './store';
+
+function errorLoading(err) {
+  console.error('Dynamic page loading failed', err);
+}
+
+function loadModule(cb) {
+  return (module) => cb(null, module.default);
+}
+
+function loadReducer(store, name) {
+  return (module) => injectAsyncReducer(store, name, module.default);
+}
 
 export default function createRoutes(store) { // eslint-disable-line
   return [
     {
       path: '/',
-      getComponent: function get(location, cb) {
-        require.ensure([], (require) => {
-          cb(null, require('HomePage').default);
-        }, 'HomePage');
+      getComponent(location, cb) {
+        Promise.all([
+          System.import('HomePage/reducer'),
+          System.import('HomePage'),
+        ]).then(modules => {
+          loadReducer(store, 'home')(modules[0]);
+          loadModule(cb)(modules[1]);
+        }).catch(errorLoading);
```

Comment:
The Homepage doesn't have a reducer associated with it in the cleaned state, so the async reducer loading here is not necessary!